python-package/lightgbm/sklearn.py

```diff
@@ -451,34 +426,26 @@ def apply(self, X, num_iteration=0):
         -------
         X_leaves : array_like, shape=[n_samples, n_trees]
         """
-        return self._Booster.predict(X,
-                                     pred_leaf=True,
-                                     num_iteration=num_iteration)
+        return self.booster_.predict(X, pred_leaf=True, num_iteration=num_iteration)
 
-    def evals_result(self):
-        """
-        Return the evaluation results.
-
-        Returns
-        -------
-        evals_result : dictionary
-        """
-        if self.evals_result_:
-            evals_result = self.evals_result_
-        else:
-            raise LightGBMError('No results found.')
-
-        return evals_result
-
-    def feature_importance(self):
-        """
-        Feature importances
+    @property
+    def booster_(self):
```

Comment:
sklearn does not have similar method, it's more like estimators_ I think.

I misunderstood this method before. The main usage of this method is to check _Booster is not None before we access to it, so it's more like a property method does. Do you think it's reasonable?